src/Latte/Engine.php

```diff
@@ -156,33 +156,62 @@ private function loadCacheFile($name, $params)
 		}
 
 		$file = $this->getCacheFile($name);
-		$handle = fopen($file, 'c+');
-		if (!$handle) {
-			throw new \RuntimeException("Unable to open or create file '$file'.");
+		if (!$this->isExpired($file, $name) && self::tryInclude($file, $params)) {
+			return;
 		}
-		flock($handle, LOCK_SH);
-		$stat = fstat($handle);
-		if (!$stat['size'] || ($this->autoRefresh && $this->getLoader()->isExpired($name, $stat['mtime']))) {
-			ftruncate($handle, 0);
-			flock($handle, LOCK_EX);
-			$stat = fstat($handle);
-			if (!$stat['size']) {
-				$code = $this->compile($name);
-				if (fwrite($handle, $code, strlen($code)) !== strlen($code)) {
-					ftruncate($handle, 0);
-					throw new \RuntimeException("Unable to write file '$file'.");
-				}
-			}
-			flock($handle, LOCK_SH); // holds the lock
+
+		if (!is_dir($this->tempDirectory)) {
+			@mkdir($this->tempDirectory); // @ - directory may already exist
```

Comment:
@JanTvrdik What about `if (@!mkdir($this->tempDirectory) && !is_dir($this->tempDirectory)) throw ... 'Unable to create temp dir'`?